src/payment/strategies/apple-pay/apple-pay-payment-strategy.ts

```diff
@@ -0,0 +1,189 @@
+import { RequestSender } from '@bigcommerce/request-sender';
+
+import { Cart } from '../../../cart';​
+import { Checkout, CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import { NotInitializedError, NotInitializedErrorType } from '../../../common/error/errors';
+import { StoreConfig } from '../../../config';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import { OrderFinalizationNotRequiredError } from '../../../order/errors';
+import { PaymentArgumentInvalidError, PaymentMethodCancelledError } from '../../errors';
+import Payment from '../../payment';
+import PaymentActionCreator from '../../payment-action-creator';
+import PaymentMethod from '../../payment-method';
+import PaymentMethodActionCreator from '../../payment-method-action-creator';
+import { PaymentInitializeOptions, PaymentRequestOptions } from '../../payment-request-options';
+import PaymentStrategy from '../payment-strategy';
+
+import ApplePaySessionFactory from './apple-pay-session-factory';
+import { assertApplePayWindow } from './is-apple-pay-window';
+
+const validationEndpoint = (bigPayEndpoint: string) => `${bigPayEndpoint}/api/public/v1/payments/applepay/validate_merchant`;
+
+interface ApplePayPromise {
+    resolve(value: InternalCheckoutSelectors | PromiseLike<InternalCheckoutSelectors>): void;
+    reject(reason?: Error): void;
+}
+​
+export default class ApplePayPaymentStrategy implements PaymentStrategy {
+    private _shippingLabel = '';
```

Comment:
```suggestion
    private _shippingLabel: string = 'Shipping';
```

Reply:
The typescript compiler is fairly good at inferring types. So do we still need this?

Reply:
They are different though, `shippingLabel: string` is different to `shippingLabel: 'Shipping'`